crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -757,18 +757,20 @@ impl CheckpointStore {
     /// A cert without contents is only stored if we have already processed
     /// internally the checkpoint. A cert with contents is processed as if
     /// it came from the internal consensus.
+    ///
+    /// Returns whether a new cert is stored locally.
```

Comment:
I think we need to extend now the doc here to state that this will process the checkpoint (even when provided with the contents = execution digests) if the transactions are already executed. So if this fails, we should do full sync to this checkpoint as this function alone will not do the execution required. 

Should we as a result reduce its scope, and have a (1) a function that updates a signed checkpoint to a certified checkpoints and (2) the function that does full sync to a checkpoint?